### PR TITLE
Document: How to use named data sources

### PR DESCRIPTION
**Pull request by john-guerra**
Adding some code examples on how to use named data sources

Please:
- [ x] Make your pull request atomic, fixing one issue at a time unless there are multiple relevant issues that would not make sense in isolation. For the latter case, please make atomic commits so we can easily review each issue.
  - Please add new commit(s) when addressing comments, so we can see easily the new changeset (instead of the whole changeset).
- [N/A] Provide a test case & update the documentation under `site/docs/`
- [N/A] Make lint and test pass. (Run `yarn test`.  If your change affects Vega outputs of some examples, re-run `yarn build` and run `yarn build:example EXAMPLE_NAME` to re-compile a specific example or `yarn build:examples` to re-compile all examples.)
- [x] Rebase onto the latest `master` branch.
- [x ] Provide a concise title that we can copy to our release note.
  - Use imperative mood and present tense.
  - Mention relevant issues. (e.g., `#1`)
- [x ] Look at the whole changeset as if you're a reviewer yourself. This will help us focus on catching issues that you might not notice yourself. (For a work-in-progress PR, you can add "[WIP]" prefix to the PR's title. When the PR is ready, you can then remove "[WIP]" and add a comment to notify us.)

Pro-Tip: https://medium.com/@greenberg/writing-pull-requests-your-coworkers-might-enjoy-reading-9d0307e93da3 is a nice article about writing a nice pull request.

### CODE REVIEW

File: site/docs/data.md [hunk 1]
```diff
@@ -71,6 +71,34 @@ For example, to create a data source named `myData`, use the following data
 }
 ```
 
+Then you can load it with vegaEmbed modying the [Vega View](https://vega.github.io/vega/docs/api/view/#data) using the insert or change methods.
```
Comment (domoritz) [inline site/docs/data.md]: The Vega view also supports changesets.

File: site/docs/data.md [hunk 2]
```diff
@@ -71,6 +71,34 @@ For example, to create a data source named `myData`, use the following data
...
+
+```js 
+const embed_opt = {"mode": "vega-lite"};
```
Comment (domoritz) [inline site/docs/data.md]: This is not necessary when you use `$schema`. Let's omit it here.

File: site/docs/data.md [hunk 3]
```diff
@@ -71,6 +71,34 @@ For example, to create a data source named `myData`, use the following data
...
+const view = vegaEmbed("#vis", spec, embed_opt)
+      ...
+			.then((res) => res.view
+        .insert("myData", [ /* some data array */])
+        .run()
+      );
+```
+
+Or if you prefer avoiding vegaEmbed
```
Comment (domoritz) [inline site/docs/data.md]: Let's remove this second example to avoid confusion.

File: site/docs/data.md (hunk 1 above)
Comment (domoritz) [reply site/docs/data.md]: Can we write this less specifically about Vega-Embed? Something along the lines of "You can use the Vega view API to load data at runtime and update the chart. Here is an example using Vega-Embed:"

File: site/docs/data.md (hunk 3 above)
Comment (domoritz) [reply site/docs/data.md]: Instead, let's link to https://vega.github.io/vega-lite/tutorials/streaming.html as an example.

File: site/docs/data.md (hunk 1 above)
Comment (john-guerra) [reply site/docs/data.md]: I think it is better to explain it first without the changeset, as the code is simpler. I also agree on the rewording

File: site/docs/data.md (hunk 2 above)
Comment (john-guerra) [reply site/docs/data.md]: ok

File: site/docs/data.md (hunk 3 above)
Comment (john-guerra) [reply site/docs/data.md]: Good idea.

**Comment by domoritz** [comment]
Thank you!

**Comment by domoritz** [comment]
Note that embed does not return a view. I added a commit to change it.

**Comment by john-guerra** [comment]
Cheers

**Pull request merged**